upload/catalog/model/extension/payment/twispay.php

```diff
@@ -1,118 +1,115 @@
+ * @version  1.0.0
+ */
+
+class ModelExtensionPaymentTwispay extends Model
+{
+    public function getMethod($address, $total)
+    {
+        $this->load->language('extension/payment/twispay');
+        $testMode = $this->config->get('twispay_testMode');
+        if (!empty($testMode)) {
+            $siteId = trim($this->config->get('twispay_staging_site_id'));
+            $privateKEY = trim($this->config->get('twispay_staging_site_key'));
+        } else {
+            $siteId = trim($this->config->get('twispay_live_site_id'));
+            $privateKEY = trim($this->config->get('twispay_live_site_key'));
+        }
+        if (empty($siteId) || empty($privateKEY)) {
+            return FALSE;
+        }
+        $method_data = array(
+            'code'       => 'twispay',
+            'title'      => $this->language->get('text_title'),
+            'terms'      =>'',
+            'sort_order' => $this->config->get('custom_sort_order')
+        );
+
+        return $method_data;
+    }
+
+    public function recurringPayments()
+    {
+        /*
+         * Used by the checkout to state the module
+         * supports recurring recurrings.
+         */
+        return TRUE;
+    }
+
+    public function checktransactions($_id=0)
+    {
+        $db_id = $this->db->escape($_id);
+        $data = $this->db->query("SELECT * FROM `".DB_PREFIX."twispay_transactions` WHERE `transactionId`='".$db_id."'");
+        if ($data->num_rows > 0) {
+            return TRUE;
+        } else {
+            return FALSE;
+        }
+    }
+
+
+    public function loggTransaction($data) {
+        $data =json_decode(json_encode($data),TRUE);
+        $data['order_id'] = $data['externalOrderId'];
+        $columns = array(
+            'order_id',
+            'status',
+            'invoice',
+            'identifier',
+            'customerId',
+            'orderId',
+            'cardId',
+            'transactionId',
+            'transactionKind',
+            'amount',
+            'currency',
+            'date',
+        );
+
+        if(!empty($data['timestamp'])) {
+            if(is_array($data['timestamp'])){
+              $data['date'] = date('Y-m-d H:i:s', strtotime($data['timestamp']['date']));
+            } else {
+              $data['date'] = date('Y-m-d H:i:s', $data['timestamp']);
+            }
+            unset($data['timestamp']);
+        }
+
+        if(!empty($data['identifier'])) {
+            $data['identifier'] = (int)str_replace('_', '', $data['identifier']);
+        }
+
+        $query = "INSERT INTO `" . DB_PREFIX . "twispay_transactions` SET ";
+        foreach($data as $key => $value) {
+            if(!in_array($key, $columns)) {
+                unset($data[$key]);
+            } else {
+                $db_value = $this->db->escape($value);
+                $query .= $key."="."'" . $db_value . "',";
+            }
+        }
+
+        $query = rtrim($query,',');
+        $this->db->query($query);
```

Comment:
Is this working? There is a column defined in the `twispay_transactions` table as `refund_date DATETIME NOT NUL` but it's not listed in the `$columns` array so there will be not value set for the column when executing the query ...

Reply:
The value of refund_date column is default setted as '0000-00-00 00:00:00'. Is working well with my mysql version, but to be sure that everything is correct i'll change the declaration from "NOT NULL" to  "NOT NULL DEFAULT '0000-00-00 00:00:00'".